### PR TITLE
upload chitter

### PR DESCRIPTION
**Pull request by DoodleDeBug**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,30 @@
...
+  - Run the query saved in the file 01_create_peeps_table.sql for each database
+
+### Running tests:
+
+  - Run `rspec`
+
+### Live View:
 
-As usual please start by forking this repo.
+  - Run `rackup`
```
Comment (willjshark) [inline README.md:26]: Great level of detail in these instructions for users

File: lib/peep.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  def self.create(peep:, name:, username:)
+
+    if ENV['ENVIRONMENT'] == 'test'
+      connection = PG.connect(dbname: 'chitter_test')
+    else
+      connection = PG.connect(dbname: 'chitter')
+    end
+
+    connection.exec("INSERT INTO peeps (peep, name, username) VALUES('#{peep}','#{name}','#{username}');")
+  end
```
Comment (willjshark) [inline lib/peep.rb:14]: You may have not made it that  far on the bookmark challenges, but connection.exec_params is best best practice compared to just connection.exec. Really interesting thing to look into if you get a chance, it basically helps stop your app being altered by SQL injections. (it's covered on step 12 of the bookmark challenge)

File: spec/features/view_peeps_spec.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+require 'pg'
+
+feature 'Viewing peeps' do
+  scenario 'A user can see recent peeps in reverse order' do
+    connection = PG.connect(dbname: 'chitter_test')
+
+    # Add the test data
```
Comment (willjshark) [inline spec/features/view_peeps_spec.rb:7]: Nice use of comments

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    erb :peep_form
+  end
+
+  post '/view_peeps' do
+    Peep.create(name: params[:name], username: params[:username], peep: params[:peep])
+    
+    redirect '/'
+  end
+
+
```
Comment (willjshark) [inline app.rb:27]: Really neat and concise code well done!

**Review by willjshark — COMMENTED** [review]
Really easy to read and logical code. I especially like your approach to commenting your code, you've labelled everything really well and your README is spot on. Only thing I would say you could work on is maybe committing more to your Github so you can see a clearer image of your step by step process. Apart from that I couldn't fault it, nice work!

File: lib/peep.rb (hunk 2 above)
Comment (DoodleDeBug) [reply lib/peep.rb:14]: Yeah I was out most of that week with tech issues, thanks I'll check it out!

**Pull request closed without merging**